WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -1421,16 +1420,30 @@ - (void)savePost
     DDLogMethod();
     [self logSavePostStats];
 
-    [self.view endEditing:YES];
-
 	__block NSString *postTitle = self.post.postTitle;
     __block NSString *postStatus = self.post.status;
     __block BOOL postIsScheduled = self.post.isScheduled;
-    
+    void (^stopEditingAndDismiss)() = ^() {
+        [self stopEditing];
+        [self.view endEditing:YES];
+        [self didSaveNewPost];
+        [self dismissEditView:YES];
+    };
+
+    NSString *hudText;
+    if (postIsScheduled) {
+        hudText = NSLocalizedString(@"Scheduling...", @"Text displayed in HUD while a post is being scheduled to be published.");
+    } else if ([postStatus isEqualToString:@"publish"]){
+        hudText = NSLocalizedString(@"Publishing...", @"Text displayed in HUD while a post is being published.");
+    } else {
+        hudText = NSLocalizedString(@"Saving...", @"Text displayed in HUD while a post is being saved as a draft.");
+    }
```

Comment:
@nheagy It does not fix this behavior if this code is broken. We definitely need to work on a state machine for this status stuff - it's super confusing.